logging/WFCORE-4807-log-routing.adoc

```diff
@@ -0,0 +1,109 @@
+= Log Routing
+:author:            James R. Perkins
+:email:             jperkins@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently any modules that log generally log onto the system log context. In some cases, for example a deployment using
+a logging profile, the loggers may end up getting created on the wrong `LogContext`. For example if `a.war` and `b.war`
+both depend on module `org.jboss.custom.module` and at least one deployment uses a logging profile whichever deployment
```

Comment:
Can you please elaborate on this example a more in analysis document? This is my current understanding of this example, is this correct?

1. `a.war` deployment uses custom logging profile and this deployment initializes `org.jboss.custom.module` module. Loggers in this custom module is configured to use custom logging profile and its `LogContext` from `a.war`
2. `b.war` deployment doesn't use any custom logging profile, so it uses default system `LogContext`. It also uses some classes from custom module, but this classes are already configured to use `LogContext` from custom profile, which is wrong.

Is this my understanding correct?

Reply:
Can you please describe on previous example, how logs would be routed with after this RFE will be implemented? I hope that I understand how logs are routed now, but I'm still not sure about the goal of this RFE.

Reply:
Sorry, I'm really not sure how I missed this. With this RFE logs would be routed to any specific log context associated with the deployment. In the above example log messages from `org.jboss.custom.module` would end up in `a.war`'s log context for anything coming from `a.war` and to the `server.log` for anything in `b.war`.

Reply:
Thank you. Can you please clarify this in a hard requirements section of the analysis directly?

Maybe (if I understand the situation correctly):
* `This means that if the TCCL is set to the deployment class loader log messages will be routed to the deployments log context.` -> `This means that if the TCCL is set to the deployment class loader log messages (from any modules called within that deployment) will be routed to the deployments log context.` + When the TCCL is set to the deployment class loader? When allow-log-routing == true? If yes, please mention this in the analysis.

Reply:
@marekkopecky I've attempted to clarify this in the second paragraph of the overview. I hope this helps. I'm trying to think of a good way to explain this and I realize it's kind of complicated and requires some knowledge of the log manager. :) It might be a bit confusing for end-users so I'll try to think of some good wording for documentation.